packages/datetime/src/dateRangeInput.tsx

```diff
@@ -128,9 +165,19 @@ export class DateRangeInput extends AbstractComponent<IDateRangeInputProps, IDat
         );
     }
 
+    public componentWillReceiveProps(nextProps: IDateRangeInputProps) {
+        if (nextProps.value !== this.props.value) {
+            const [selectedStart, selectedEnd] = this.getInitialRange(nextProps);
+            this.setState({ selectedStart, selectedEnd });
+        }
+        super.componentWillReceiveProps(nextProps);
```

Comment:
`super` should typically be the first thing you do, if you're gonna do it. TS actually enforces this for constructors.